src/main/java/com/faforever/client/io/FeaturedModFileCacheService.java

```diff
@@ -0,0 +1,118 @@
+package com.faforever.client.io;
+
+import com.faforever.client.api.dto.FeaturedModFile;
+import com.faforever.client.preferences.PreferencesService;
+import com.faforever.client.task.ResourceLocks;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.stream.Stream;
+
+
+@Service
+@Slf4j
+public class FeaturedModFileCacheService implements InitializingBean {
+  private final Path cacheDirectory;
+  private final int cacheLifeTimeInDays;
+
+  public FeaturedModFileCacheService(PreferencesService preferencesService) {
+    this.cacheDirectory = preferencesService.getCacheDirectory();
+    this.cacheLifeTimeInDays = preferencesService.getPreferences().getCacheLifeTimeInDays();
```

Comment:
If they are cached here changes to this variables are later not reflected. While this does not matter for cacheLifeTime it might be for the cache driectory(even that is also unlikely)